### DISCUSSION TITLE
MPS overlaps with specified determinants

### DISCUSSION THREAD

**Discussion by ghb24** [block-hczhai/block2-preview#86] [Q&A] [answered]
Hi Block2Devs,
I was wondering whether it was possible for Block2 to compute the overlap with specified determinants? I can see in the docs that it can find all overlaps with determinants/CSFs greater than a specific threshold, but is it possible to find e.g. the amplitudes on all double excitations of a reference determinant for tailored-CCSD? Potentially also with a threshold imposed?
Many thanks in advance

**Comment by hczhai** [comment, marked as answer]
Thanks for your interest in using block2.

This can be done using the ``given_dets`` argument in ``DMRGDriver.get_csf_coefficients``, see https://block2.readthedocs.io/en/latest/api/pyblock2.html#pyblock2.driver.core.DMRGDriver.get_csf_coefficients. Detailed explanation can be found in https://github.com/block-hczhai/block2-preview/discussions/70#discussioncomment-7822193. When working in the ``SZ`` mode, you need ``given_dets=['2ba0ba', '0ab2ba']`` (for example).

Minimal ``block2`` version that supports this feature is 0.5.3rc5. You can install this version (or any later version) using
```
pip install block2==0.5.3rc5 --extra-index-url=https://block-hczhai.github.io/block2-preview/pypi/
```

**Reply by ghb24** [reply]
Fantastic - many thanks for the quick response!
